couchbase/datadog_checks/couchbase/couchbase.py

```diff
@@ -402,3 +410,83 @@ def extract_seconds_value(self, value):
+            self.log.warning(str(e))
+            self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.CRITICAL, self._tags, msg)
+            try:
+                # Error handling in case Couchbase changes their versioning format
+                if self._version and int(self._version.split(".")[0]) < 7:
+                    self.log.warning(
+                        "Index Stats Metrics are only available in Couchbase version 7+. Detected version: %",
+                        self._version,
+                    )
+            except Exception as e:
+                self.log.warning(str(e))
+            return
+
+        self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.OK, self._tags)
+
+        for keyspace in data:
+            if keyspace == "indexer":
+                # The indexer object provides metric about the index node
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_index_node_metrics(mname, mval, self._tags)
+            else:
+                index_tags = self._extract_index_tags(keyspace) + self._tags
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_per_index_metrics(mname, mval, index_tags)
+
+    def _extract_index_tags(self, keyspace):
+        # Index Keyspaces can come in different formats:
+        # partition, bucket:index_name, bucket:collection:index_name, bucket:scope:collection:index_name
+        # For variations missing the scope and collection, they refer to the default scope and collection respectively
+        # https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/createprimaryindex.html#keyspace-ref
+        tag_arr = keyspace.split(":")
+        if len(tag_arr) == 2:
+            bucket, index_name = tag_arr
+            scope, collection = ["default", "default"]
+        elif len(tag_arr) == 3:
+            bucket, collection, index_name = tag_arr
+            scope = 'default'
+        elif len(tag_arr) == 4:
+            bucket, scope, collection, index_name = tag_arr
+        else:
+            # Catch all incase the keyspace has either none or 3 or more separators(':')
+            # There is a documented example of partition-num being a possible keyspace:
+            # https://docs.couchbase.com/server/current/rest-api/rest-index-stats.html#_get_index_stats
+            # But we shouldn't encounter this since we don't query the index api with the needed params
+            # (Version 1.19.0+ of the Couchbase check)
+            formatted_tags = []
+            self.log.debug("Unable to extract tags from keyspace: %s", keyspace)
+            return formatted_tags
+
+        formatted_tags = [
+            'bucket:{}'.format(bucket),
+            'scope:{}'.format(scope),
+            'collection:{}'.format(collection),
+            'index_name:{}'.format(index_name),
+        ]
+        return formatted_tags
+
+    def _submit_index_node_metrics(self, mname, mval, tags):
```

Comment:
```suggestion
    def _submit_index_node_metrics(self, mname, mval):
```

this function is only used once and passes `self.tags` which can be called directly

Reply:
Make sense. I added your change and attach tags via `self._tags` in the submit function.